File: src/main/java/com/wanted/preonboarding/ticket/domain/entity/EnableDisableConverter.java
package com.wanted.preonboarding.ticket.domain.entity;

import jakarta.persistence.AttributeConverter;

public class EnableDisableConverter implements AttributeConverter<Boolean,String> {

    @Override
    public String convertToDatabaseColumn(Boolean attribute) {
        return (attribute != null && attribute) ? "enable" : "disable";
    }

    @Override
    public Boolean convertToEntityAttribute(String dbData) {
        return "enable".equalsIgnoreCase(dbData);
    }
}

File: src/main/kotlin/com/wanted/preonboarding/ticket/infra/entity/EnableDisableConverter.kt
package com.wanted.preonboarding.ticket.infra.entity

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class EnableDisableConverter : AttributeConverter<Boolean, String> {
    override fun convertToDatabaseColumn(attribute: Boolean): String {
        return if (attribute) "enable" else "disable"
    }

    override fun convertToEntityAttribute(dbData: String): Boolean {
        return dbData == "enable"
    }
}
